src/components/GlobalPreferences/Notifications/notification-service-api.js

```diff
@@ -0,0 +1,276 @@
+import {
+  NOTIFICATION_SERVICE_ACCOUNT,
+  NOTIFICATION_SERVICE_LOGIN,
+  NOTIFICATION_SERVICE_VERIFY,
+  NOTIFICATION_SERVICE_SUBSCRIPTIONS,
+  API_MESSAGE_EXPIRED_TOKEN,
+  ExpiredTokenError,
+  UnauthorizedError,
+} from './constants'
+import { getEthNetworkType } from '../../../local-settings'
+
+const isAuthTokenExpired = response =>
+  response.statusCode === 401 && response.message === API_MESSAGE_EXPIRED_TOKEN
+
+const isUnauthorized = rawResponse => rawResponse.status === 401
+
+export const login = async ({ email, dao, network }) => {
+  try {
+    const rawResponse = await fetch(NOTIFICATION_SERVICE_LOGIN, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ email, dao, network }),
+    })
+    if (!rawResponse.ok) {
+      throw new Error(rawResponse.statusText)
+    }
+  } catch (e) {
+    console.error(e)
+    throw e
+  }
+}
+
+// Verify the short lived email token and fetch a long lived token
+export async function verifyEmailToken(shortLivedToken) {
```

Comment:
Just a heads up, `https://jwt.io/` says the signature is not valid for a token I received:

![image](https://user-images.githubusercontent.com/3072458/64104289-450ea980-cd74-11e9-9054-3f553cb37e29.png)

Reply:
That's because the `secret` is required to verify the signature. 

As a side note, we use JWT with symmetric encryption for simplicity and because the token is used by a single service. If we used JWT with asymmetric encryption, we could use the public key to verify the signature without needing access to the private key.